app/src/main/java/com/example/hungrywolfes/ui/detailsScreen/DetailsFragment.kt

```diff
@@ -0,0 +1,91 @@
+package com.example.hungrywolfes.ui.detailScreen
+
+import android.annotation.SuppressLint
+import android.app.Activity
+import android.content.Context
+import android.os.Bundle
+import android.util.Log
+import androidx.fragment.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.view.inputmethod.InputMethodManager
+import androidx.core.content.ContextCompat
+import androidx.fragment.app.viewModels
+import androidx.navigation.fragment.findNavController
+import androidx.recyclerview.widget.DividerItemDecoration
+import androidx.recyclerview.widget.RecyclerView
+import com.bumptech.glide.Glide
+import com.example.hungrywolfes.DetailBinding
+import com.example.hungrywolfes.R
+import com.example.hungrywolfes.ui.DetailsAdapter
+import com.example.hungrywolfes.ui.detailsScreen.DetailsViewModel
+
+private const val TAG = "DetailFragment"
+
+class DetailsFragment : Fragment() {
+    private lateinit var binding: DetailBinding
+    private lateinit var idMealHome: String
+    private lateinit var idMealSearch: String
+    private val viewModel: DetailsViewModel by viewModels()
+    private val detailsAdapter = DetailsAdapter()
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        binding = DetailBinding.inflate(inflater, container, false).apply {
+            lifecycleOwner = viewLifecycleOwner
+            viewModel = this@DetailsFragment.viewModel
+        }
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        getArgumentsFromSearchScreen()
+        getArgumentsFromHomeScreen()
+        setupRecyclerView()
+        setupObservers()
+    }
+
+    private fun getArgumentsFromHomeScreen() {
+        arguments?.let { idMealHome = it.getString("idMeal").toString() }
+        viewModel.getDetailsMeal(idMealHome.toInt())
+    }
+    private fun getArgumentsFromSearchScreen(){
+        arguments?.let{idMealSearch=it.getString("idMeal").toString()}
```

Comment:
1. Don't forget to format code. CTRL+ALT+L
2. Since you added the safe-args, have you tried to get the args inside the fragment like it is mentioned in the [docs](https://developer.android.com/guide/navigation/navigation-pass-data#Safe-args)?